server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -461,15 +462,15 @@ ClusterService getClusterService() {
         return clusterService;
     }
 
-    Map<String, DiscoveryExtensionNode> getExtensionIdMap() {
+    public Map<String, DiscoveryExtensionNode> getExtensionIdMap() {
```

Comment:
More reasons to drop the ApplicationManager

Reply:
The application manager can be moved out of this PR if you prefer, it will be required in order to resolve Plugins and Applications in a similar way like in https://github.com/opensearch-project/OpenSearch/pull/8526. Do you prefer I take it out for now?